src/index.html

```diff
@@ -11,7 +11,177 @@
+            name="age"
+            min="1"
+            max="100"
+            value="12"
+          >
+        </label><br>
+
+        <label class="form-field">
+          Full date of birth:
+          <input
+            class="distanceboth"
+            type="date"
+            name="date"
+          >
+        </label><br>
+
+        <label class="form-field">
+          I accept the term of the agreement
+          <input
+            type="checkbox"
+            name="agreement"
+          >
+        </label>
+      </fieldset>
+
+      <fieldset class="distanceGroups">
+        <legend>Registration</legend>
+
+        <label class="form-field">
+          E-mail:
+          <input
+            type="email"
+            name="email"
+            placeholder="email@example.com"
+            required
+          >
+        </label><br>
+
+        <label class="form-field">
+          Password:
+          <input class="distance"
+            type="password"
+            name="password"
+            minlength="4"
+            maxlength="8"
+          >
+        </label>
+      </fieldset>
+
+      <fieldset class="distanceGroups">
+        <legend>An interesting fact about you!</legend>
+
+        <label class="form-field">
+          Do you love cats?
+          <label>
+            <input
+              type="radio"
+              name="cats"
+              value="Yes"
+            >
+            Yes
+          </label>
+          <label>
+            <input
+              type="radio"
+              name="cats"
+              value="No"
+            >
+            No
+          </label>
+        </label><br>
+
+        <label class="form-field">
+          What is your favorite color?
+          <input
+            class="distance"
+            type="color"
+            name="color"
+          >
+        </label><br>
+
+        <label class="form-field distance">
+          What time do you go to bed?
+          <input
+            class="distance"
+            type="time"
+            name="time"
+          >
+        </label><br>
+
+        <label class="form-field">
+          What are your favorite brand of cars?
+          <select
+            class="distanceboth"
+            name="brand"
+            multiple
+          >
+            <option value="BMW">BMW</option>
+            <option value="Audi">Audi</option>
+            <option value="Lada">Lada</option>
+          </select>
+        </label><br>
+
+        <label class="form-field">
+          How do you rate our work?
+          <input
+            type="range"
+            name="rate"
+            value="0"
+            min="0"
+            max="10"
+            step="1"
+          >
+        </label>
+      </fieldset>
+
+      <fieldset class="distanceGroups">
+        <legend>Additional info:</legend>
+
+        <label>
+          Comments:
+          <textarea
+            name="comments"
+            cols="20"
+            rows="2"
+          >
+          </textarea>
```

Comment:
use this trick to avoid empty space when focusing on textarea
```suggestion
          ></textarea>
```